File: python/examples/fork/fork.py
import os
import time

name = "common"

def child():
    time.sleep(1)
    print(f"In Child ({name}).")
    print(f"In Child PID: {os.getpid()} PPID: {os.getppid()}")

    time.sleep(5)
    exit(3)

def parent(child_pid):
    time.sleep(1)
    print(f"In Parent ({name}) The child is: {child_pid}")
    print(f"In Parent PID: {os.getpid()} PPID: {os.getppid()}")
    r = os.wait()
    print(r)

print(f'{os.getpid()} - start running')
pid = os.fork()
print(f'my pid: {os.getpid()} pid received from fork: {pid}')
if pid == 0:
    child()
else:
    parent(pid)

File: perl/examples/tasks/fork.pl
use strict;
use warnings;

main();

sub main {
    my $pid = fork();
    die "Could not fork" if not defined $pid;
    if (not $pid) {
        sleep 1;
        my $parent = getppid();
        print "In Child:  PID: $$ from parnet: $parent\n";;
        exit;
    }
    sleep 1;
    print "In Parent: PID: $$ Child: $pid\n";
    my $finished = wait();
    my $exit_code = $? / 256;
    print "Child finised: $finished Exit code: $exit_code\n";
}

